src/classes/COFF_Affiliation_TDTM.cls

```diff
@@ -146,9 +146,7 @@ public class COFF_Affiliation_TDTM extends TDTM_Runnable {
                         //   --> update Affiliation to Dpt to point it to the new Faculty
                         } else if(oldOff.Faculty__c != null && newOff.Faculty__c != null && oldOff.Faculty__c != newOff.Faculty__c) {
                             UTIL_Debug.debug('****Faculty member has been changed');
-                            if (UTIL_CustomSettingsFacade.courseConnectionsEnabled()) {
```

Comment:
Yes, this is why it is causing the error. The system is not preventing users from using course connections when it is disabled by custom setting.